### PR TITLE
Deprecate storage class overrides

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Task

### Description

This PR deprecates the storage class overrides in storage configuration as approved in [Strimzi Proposal no. 80](https://github.com/strimzi/proposals/blob/main/080-deprecation-and-removal-of-storage-overrides.md). It also updates the documentation accordingly. The warnings when the overrides are used are issued automatically for the deprecations. So there is no special code or test for them.

### Checklist

- [x] Update documentation
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Update CHANGELOG.md

### CODE REVIEW

File: api/src/main/java/io/strimzi/api/kafka/model/kafka/PersistentClaimStorage.java [hunk 1]
```diff
@@ -114,6 +115,9 @@ public void setDeleteClaim(boolean deleteClaim) {
     @Description("Overrides for individual brokers. " +
             "The `overrides` field allows to specify a different configuration for different brokers.")
     @JsonInclude(JsonInclude.Include.NON_NULL)
+    @Deprecated
+    @DeprecatedProperty(description = "The storage overrides for individual brokers are deprecated and will be removed in the future. " +
+            "Please use multiple KafkaNodePool custom resources with different storage classes instead.")
```
Comment (PaulRMellor) [inline api/src/main/java/io/strimzi/api/kafka/model/kafka/PersistentClaimStorage.java:120]:
```suggestion
            "Please use multiple `KafkaNodePool` custom resources with different storage classes instead.")
```

File: api/src/main/java/io/strimzi/api/kafka/model/kafka/PersistentClaimStorage.java [hunk 2]
```diff
@@ -114,6 +115,9 @@ public void setDeleteClaim(boolean deleteClaim) {
     @Description("Overrides for individual brokers. " +
             "The `overrides` field allows to specify a different configuration for different brokers.")
```
Comment (PaulRMellor) [inline api/src/main/java/io/strimzi/api/kafka/model/kafka/PersistentClaimStorage.java:116]:
```suggestion
            "The `overrides` field allows you to specify a different configuration for different brokers.")
```

File: documentation/modules/configuring/ref-storage-persistent.adoc [hunk 3]
```diff
@@ -124,6 +124,8 @@ storage:
 
 == Storage class overrides
 
+WARNING: Storage class overrides are deprecated and will be removed in the future. Please use Kafka Node Pool resources instead.
```
Comment (PaulRMellor) [inline documentation/modules/configuring/ref-storage-persistent.adoc:127]:
```suggestion
WARNING: Storage class overrides are deprecated and will be removed in the future. As a replacement, use `KafkaNodePool` resources instead.
```

File: documentation/modules/configuring/ref-storage-persistent.adoc [hunk 4]
```diff
@@ -190,6 +192,19 @@ As a result of the configured `overrides` property, the volumes use the followin
 The `overrides` property is currently used only to override the storage `class`. 
 Overrides for other storage configuration properties is not currently supported.
 
+=== Migrating from storage class overrides to Kafka Node Pools
```
Comment (PaulRMellor) [inline documentation/modules/configuring/ref-storage-persistent.adoc:195]:
```suggestion
=== Migrating from storage class overrides to node pools
```

File: documentation/modules/configuring/ref-storage-persistent.adoc [hunk 5]
```diff
@@ -190,6 +192,19 @@ As a result of the configured `overrides` property, the volumes use the followin
...
+
+Storage class overrides are deprecated and will be removed in the future.
+If you use storage class overrides, you should use Kafka Node Pools instead.
```
Comment (PaulRMellor) [inline documentation/modules/configuring/ref-storage-persistent.adoc:198]:
```suggestion
If you are using storage class overrides, we encourage you to transition to using node pools instead.
```

File: documentation/modules/configuring/ref-storage-persistent.adoc [hunk 6]
```diff
@@ -190,6 +192,19 @@ As a result of the configured `overrides` property, the volumes use the followin
...
+To migrate the existing configuration, you should follow these steps:
```
Comment (PaulRMellor) [inline documentation/modules/configuring/ref-storage-persistent.adoc:199]:
```suggestion
To migrate the existing configuration, follow these steps:
```

**Review by PaulRMellor — APPROVED** [review]
Looks good, thanks. I left a few suggestions.

**Review by ppatierno — APPROVED** [review]
LGTM.

**Pull request merged**